File: backend/src/main/java/hu/szakdoga/backend/forum/data/dto/ForumDTO.java
package hu.szakdoga.backend.forum.data.dto;

import java.util.List;

public class ForumDTO {
    public final Long id;
    public final String name;
    public final String description;
    public final Long universityId;
    public final List<Long> facultyIds;
    public final List<Long> majorIds;

    public ForumDTO(Long id, String name, String description, Long universityId, List<Long> facultyIds, List<Long> majorIds) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.universityId = universityId;
        this.facultyIds = facultyIds;
        this.majorIds = majorIds;;
    }
}

File: frontend/src/app/shared/model/forum/forum.dto.ts
export class ForumDto {
    public readonly id: number | null;
    public readonly name: string;
    public readonly universityId: number;
    public readonly majorIds: number[];
    public readonly facultyIds: number[];
    public readonly description: string;
    public readonly approved: boolean;


	constructor(name: string, universityId: number, majorIds: number[], facultyIds: number[], description: string, approved: boolean, id: number | null = null){
        this.id = id;
        this.name = name;
        this.universityId = universityId;
        this.majorIds = majorIds;
        this.description = description;
        this.facultyIds = facultyIds;
        this.approved = approved;
    }
}